config/initializers/carrierwave.rb

```diff
@@ -34,17 +34,28 @@ module CarrierWave
   module Configuration
     def self.configure_fog!(credentials: OpenProject::Configuration.fog_credentials,
                             directory: OpenProject::Configuration.fog_directory,
+                            host: OpenProject::Configuration.fog_host,
+                            endpoint: OpenProject::Configuration.fog_endpoint,
                             public: false)
 
       # Ensure that the provider AWS is uppercased
       provider = credentials[:provider] || 'AWS'
       if [:aws, 'aws'].include? provider
-        provider = 'AWS'
+        credentials[:provider] = 'AWS'
+      end
+
+      # Allow setting a custom host and endpoint that will be used in stead of S3 endpoints
+      if host.present?
+        credentials[:host] = host
```

Comment:
@oliverguenther IMHO it would be simpler (and less code) to keep the host and endpoint configuration in the credentials. I.e. if you want to configure it you should have to set `OPENPROJECT_FOG_CREDENTIALS_HOST` rather than `OPENPROJECT_FOG_HOST`.  Since these options actually are part of the fog credentials this would be more consistent.

And then we could just add `OpenProject::Configuration.fog_credentials[:host]` to the remote storage hosts.

What do you think?